### PR TITLE
As a web coordinator, I would like a web template in Cascade that allows me to publish landing pages that are unique to my advertising campaigns.

### PR DESCRIPTION
**Pull request by tatwell**
Opening early for our pre-mortem code review tomorrow. Not ready for final review yet.

User story: https://trello.com/c/HceaZ8fY/56-as-a-web-coordinator-i-would-like-a-web-template-in-cascade-that-allows-me-to-publish-landing-pages-that-are-unique-to-my-advert

### CODE REVIEW

File: app/data_definitions/from_cascade/ad_landing.xml [hunk 1]
```diff
@@ -0,0 +1,130 @@
+<system-data-structure>
```
Comment (tatwell) [inline app/data_definitions/from_cascade/ad_landing.xml]: Copied directly from Dev Cascade. Now under version control. However, any changes in Cascade have to be copied over. So a bit of a double accounting issue.

File: app/data_definitions/ad_landing.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      ad_landing = DataDefinitions::AdLanding.new
+
+      DataDefinitions::AdLanding::DEFAULTS.each do |xpath, value|
+        node = ad_landing.document.at_xpath(xpath)
+        raise "Node at xpath #{xpath} not found." unless node
+        node.content = value
+      end
+
+      ad_landing
+    end
```
Comment (tatwell) [inline app/data_definitions/ad_landing.rb]: Additional profiles could be added here. For example, `self.law` or `self.dodge`.

File: app/views/_cascade/formats/modular/ad_landing/widgets/_masthead_hero.html.erb [hunk 3]
```diff
@@ -0,0 +1,99 @@
+<%#
```
Comment (tatwell) [inline app/views/_cascade/formats/modular/ad_landing/widgets/_masthead_hero.html.erb]: Would be nice if we could just copy over the Cascade format directly. But we'd need a Velocity parser or transpiler. Volunteers?

File: app/data_definitions/base.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    attr_accessor :document
+
+    def initialize
+      xml_file = format('%s.xml', self.class.name.underscore.split('/').last)
+      xml_path = File.join(Rails.root, 'app/data_definitions/from_cascade', xml_file)
+      @document = Nokogiri::XML(File.read(xml_path)) if File.exist?(xml_path)
+    end
+
+    # rubocop:disable Rails/OutputSafety
+    def get_child_value(field, unescaped=false)
```
Comment (tatwell) [inline app/data_definitions/base.rb]: Added this method to replace field-specific helper methods in data definition classes. It also better echoes what Cascade itself does in velocity templates.

**Comment by mirabelrice** [comment]
looks good, 👍

**Pull request merged**